src/main/java/org/folio/search/cql/CqlTermQueryConverter.java

```diff
@@ -143,11 +152,24 @@ private static Map<String, TermQueryBuilder> getTermQueryProvidersAsMap(List<Ter
 
   private void validateIndexFormat(String index, CQLTermNode termNode) {
     var value = termNode.getTerm();
-    if (index.equals("date") && !isDate(value, STRICT_DATE_PATTERN, true)) {
+    if (index.equals("date") && !isValidDate(value)) {
       throw new ValidationException("Invalid date format", termNode.getIndex(), value);
     }
   }
 
+  private boolean isValidDate(String value) {
+    boolean isValidDate = false;
+    for (DateTimeFormatter dateFormat : SUPPORTED_DATE_FORMATS) {
+      try {
+        dateFormat.parse(value);
+        isValidDate = true;
+      } catch (Exception ignored) {
+        // do nothing
+      }
+    }
+    return isValidDate;
```

Comment:
I assume when the first parses is successfully applied then there is no need to check the rest of them
 ```suggestion
    for (DateTimeFormatter dateFormat : SUPPORTED_DATE_FORMATS) {
      try {
        dateFormat.parse(value);
        return true;
      } catch (Exception ignored) {
        // do nothing
      }
    }
  return false;
```